packages/grafana-data/src/types/plugin.ts

```diff
@@ -23,6 +23,12 @@ export enum PluginSignatureStatus {
   unsigned = 'unsigned', // no MANIFEST file
 }
 
+export enum PluginErrorCode {
+  invalid = 'invalid',
+  modified = 'modified',
```

Comment:
are these about the signature?  lets make sure to scope the names to include that.
```suggestion
  invalidSignature = 'invalidSignature',
  signatureModified = 'signatureModified',
```

Other error stats we will want to expose will be:
* plugin health check is failing
* plugin is in some crash loop and we won't run it anymore
* (maybe) 'unavailable' or something like that -- when we expect an external gRPC connection that has not yet connected

Reply:
Makes sense. Will need them to match up with the backend enums too.

Regarding the other error states - they sound reasonable too. Can see them having a place here when necessary